build.gradle

```diff
@@ -92,8 +92,22 @@ subprojects {
     tasks.check.dependsOn(tasks.jacocoTestReport)
     tasks.check.dependsOn(javadoc)
 
-    if (System.env.CIRCLE_TEST_REPORTS) {
-        test.reports.junitXml.destination = new File(System.env.CIRCLE_TEST_REPORTS, it.getName())
+    if (System.env.CIRCLECI) {
+        test.reports.junitXml.destination = new File(System.env.TESTS_DIR, it.getName())
+
+        task copyCiArtifactsToCircle {
```

Comment:
For tests, the change is analogous -- before, this project also had logic to test for CircleCI and wrote tests into that directory.

For artifacts, previously this `circle.yml` hard-coded the locations to copy in the `circle.yml`, which was fragile (case in point, two of the newer subprojects are missing).